VocaDbModel/Service/Search/SongSearch/SongSearch.cs

```diff
@@ -101,14 +104,14 @@ private SearchTextQuery ProcessAdvancedSearch(SearchTextQuery textQuery, SongQue
 
 			if (artistNames.Any())
 			{
-				queryParams.ArtistNames = artistNames;
+				queryParams = queryParams with { ArtistNames = artistNames };
 			}
 
 			var words = parsed.GetValues("").ToArray();
 
 			if (words.Any())
 			{
-				queryParams.Common.TextQuery = new SearchTextQuery(textQuery.Query, NameMatchMode.Words, textQuery.OriginalQuery, words);
+				queryParams = queryParams with { Common = queryParams.Common with { TextQuery = new SearchTextQuery(textQuery.Query, NameMatchMode.Words, textQuery.OriginalQuery, words) } };

```

Comment:
@riipah Does it make sense to make this (and all other query params) immutable? I'm not sure if this could be a performance issue.

Reply:
Alright, thanks!